pkg/factories/rune_factory.go

```diff
@@ -23,48 +23,66 @@ import (
 )
 
 type DefautRunEFactory struct {
-	knSourceParams *types.KnSourceParams
-	knSourceClient types.KnSourceClient
+	knSourceParams        *types.KnSourceParams
+	knSourceClientFactory types.ClientFactory
+	knSourceClient        types.KnSourceClient
```

Comment:
The idea was to allow the `RunEFactory` to create the client instead of passing it.